### PR TITLE
EZP-31571: Content tree view in UDW

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-31571
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

BC: changed parameters for loadSubtree function to object: https://github.com/ezsystems/ezplatform-admin-ui/commit/810d10b7eb0c613f4fe465b24bf154df0bd04cd9#diff-d555737827e39c2e1490479db6f96e76R31

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by GrabowskiM** [comment]
Selection_026
Selection_028
Selection_029

File: src/bundle/ui-dev/src/modules/content-tree/content.tree.module.js [hunk 1]
```diff
@@ -156,6 +166,12 @@ export default class ContentTreeModule extends Component {
     }
 
     readSubtree() {
+        const { readSubtree } = this.props;
+
+        if (readSubtree) {
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/content-tree/content.tree.module.js:171]: I would check if it is a function

File: src/bundle/ui-dev/src/modules/content-tree/services/content.tree.service.js [hunk 2]
```diff
@@ -28,8 +28,13 @@ export const loadLocationItems = ({ siteaccess }, parentLocationId, callback, li
         .catch(showErrorNotification);
 };
 
-export const loadSubtree = ({ token, siteaccess }, subtree, callback) => {
-    const request = new Request(`${ENDPOINT_LOAD_SUBTREE}`, {
+export const loadSubtree = ({ restInfo: { token, siteaccess }, subtree, sortClause, sortOrder }, callback) => {
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/content-tree/services/content.tree.service.js:31]:
```suggestion
export const loadSubtree = ({ token, siteaccess, subtree, sortClause, sortOrder }, callback) => {
```

Do not pass `restInfo` object just pass the `token` and `siteaccess` this is more consistent with other places.

File: src/bundle/ui-dev/src/modules/content-tree/services/content.tree.service.js [hunk 3]
```diff
@@ -28,8 +28,13 @@ export const loadLocationItems = ({ siteaccess }, parentLocationId, callback, li
...
+    let path = ENDPOINT_LOAD_SUBTREE;
+    if (sortClause && sortOrder) {
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/content-tree/services/content.tree.service.js:33]:
```suggestion

    if (sortClause && sortOrder) {
```

File: src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                const isContainer = contentTypeInfo.isContainer;
+                const isNotSelectable =
+                    (containersOnly && !isContainer) || (allowedContentTypes && !allowedContentTypes.includes(contentTypeInfo.identifier));
+
+                setMarkedLocationId(locationId);
+                dispatchLoadedLocationsAction({ type: 'CUT_LOCATIONS', locationId: markedLocationId });
+                dispatchLoadedLocationsAction({ type: 'SET_LOCATIONS', data: locationsMap });
+
+                if (!multiple && !isNotSelectable) {
+                    dispatchSelectedLocationsAction({ type: 'CLEAR_SELECTED_LOCATIONS' });
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js:69]: You can use `REPLACE_SELECTED_LOCATIONS`

File: src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        );
+    };
+    const readSubtree = () => {
+        const tree = [];
+        let leafs = tree;
+
+        loadedLocationsMap.forEach((location) => {
+            leafs.push({
+                children: [],
+                limit: 30,
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js:82]: Why limit is hardcoded to 30?

File: src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                    dispatchSelectedLocationsAction({ type: 'ADD_SELECTED_LOCATION', location });
+                }
+            }
+        );
+    };
+    const readSubtree = () => {
+        const tree = [];
+        let leafs = tree;
+
+        loadedLocationsMap.forEach((location) => {
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js:79]: This whole method deserves to be refactored.

File: src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    };
+    const currentLocationPath = locationData && locationData.location ? locationData.location.pathString : '/1/';
+    const locationsLoaded = loadedLocationsMap.length > 1 || (loadedLocationsMap.length === 1 && loadedLocationsMap[0].subitems.length > 0);
+    const contentTreeVisible = (markedLocationId !== null && locationsLoaded) || markedLocationId === null;
+
+    return (
+        <div className="c-tree">
+            {contentTreeVisible && (
+                <ContentTreeModule
+                    userId={14}
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js:100]: Why user id is hardcoded to admin?

File: src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            },
+            (locationsMap) => {
+                const { location } = locationsMap[locationsMap.length - 1];
+                const contentTypeInfo = contentTypesMap[location.ContentInfo.Content.ContentType._href];
+                const isContainer = contentTypeInfo.isContainer;
+                const isNotSelectable =
+                    (containersOnly && !isContainer) || (allowedContentTypes && !allowedContentTypes.includes(contentTypeInfo.identifier));
+
+                setMarkedLocationId(locationId);
+                dispatchLoadedLocationsAction({ type: 'CUT_LOCATIONS', locationId: markedLocationId });
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js:65]: Why do you `CUT` and in next line you do `SET` which replace the whole `loadedLocations`?

File: src/bundle/ui-dev/src/modules/universal-discovery/bookmarks.tab.module.js [hunk 9]
```diff
@@ -30,6 +31,7 @@ const BookmarksTabModule = () => {
     const views = {
         grid: <GridView itemsPerPage={tabsConfig.bookmarks.itemsPerPage} />,
         finder: <Finder itemsPerPage={tabsConfig.bookmarks.itemsPerPage} />,
+        tree: <TreeView itemsPerPage={tabsConfig.browse.itemsPerPage} />,
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/universal-discovery/bookmarks.tab.module.js:34]:
```suggestion
        tree: <TreeView itemsPerPage={tabsConfig.bookmarks.itemsPerPage} />,
```
?

**Pull request merged**